src/components/Welcome.js

```diff
@@ -1,7 +1,46 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { Link } from 'react-router-dom';
+import { db } from '../lib/firebase';
 
 export default function Welcome() {
+  const [userInputToken, setUserInputToken] = useState('');
+
+  //references the collection where the items are stored
+  let itemsRef = db.collection('items');
+
+  // Handler function for setting the User Input Token state based on user input
+  const handleChange = (event) => {
+    event.preventDefault();
+    setUserInputToken(event.target.value);
+  };
+
+  //Checks to see if userInputToken exists in db and adds user to existing list
+  const handleJoinList = (event) => {
+    event.preventDefault();
+
+    itemsRef
+      .where('userToken', '==', userInputToken)
+      .limit(1)
+      .get()
+      .then(function (querySnapshot) {
+        // if document is empty/ userToken != userInputToken
+        //console log error message
+        if (querySnapshot.empty) {
+          console.log('User Token Does Not Exist');
+          alert('User Token Does Not Exist. Please Try Again');
```

Comment:
I think it would be better for us to show an error inline after the form is submitted.
Similar to the errors you get after your password is wrong, maybe a <span> with text color red instead of the alert.

Reply:
That makes sense! We already took out the console.log to not duplicate - but we can notify the user with red. @audracodes Want to work on this later today?

Reply:
@indiragp Do you have any suggestions for best error handling with React hooks? I'm looking up a few different ways.

Reply:
### I would use a 
const [error, setError] = useState("");
and then set the error within a useEffect, or update it with setError in this function.